scopesim/effects/metis_lms_trace_list.py

```diff
@@ -164,6 +167,90 @@ def make_spectral_traces(self):
 
         self.spectral_traces = spec_traces
 
+    def rectify_cube(self, hdulist, xi_min=None, xi_max=None, interps=None,
+                     **kwargs):
+        """
+        Rectify an IFU observation into a data cube
+
+        The HDU list (or fits file) must have been created with the
+        present OpticalTrain (or an identically configured one).
+
+        Parameters
+        ----------
+        hdulist : str or fits.HDUList
+           an ifu observation created with the present OpticalTrain
+        xi_min, xi_max : float [arcsec]
+           Spatial limits of the image slicer on the sky. For METIS LMS,
+           these values need not be provided by the user.
+        interps : list of interpolation functions
+           If provided, there must be one for each image extension in `hdulist`.
+           The functions go from pixels to the images and can be created with,
+           e.g., RectBivariateSpline.
+        """
+        try:
+            inhdul = fits.open(hdulist)
+        except TypeError:
+            inhdul = hdulist
+
+        # Create interpolation functions
+        if interps is None:
+            logging.info("Computing interpolation functions")
+            interps = make_image_interpolations(inhdul, kx=1, ky=1)
+
+        # Create a common wcs for the rectification
+        dwave = from_currsys("!SIM.spectral.spectral_bin_width")
+        xi_min = np.min(self.slicelist['left'])
+        xi_max = np.max(self.slicelist['right'])
+        wave_min = self.meta['wave_min']
+        wave_max = self.meta['wave_max']
+        pixscale = self.meta['pixel_scale']
+        naxis1 = int((xi_max - xi_min) / pixscale) + 1
+        naxis2 = len(self.spectral_traces)
+        naxis3 = int((wave_max - wave_min)/dwave) + 1
+        logging.debug(f"Cube: {naxis1}, {naxis2}, {naxis3}")
+        logging.debug(f"Xi: {xi_min}, {xi_max}")
+        logging.debug(f"Wavelength: {wave_min}, {wave_max}")
```

Comment:
```suggestion
        logging.debug("Cube: %d, %d, %d", naxis1, naxis2, naxis3)
        logging.debug("Xi: %g, %g", xi_min, xi_max)
        logging.debug("Wavelength: %g, %g", wave_min, wave_max)
```
Just to be pedantic :P 

I used `%g`, which is flexible. However, if you know what sort of values to expect and what precision you want/need, that could be replaced with something like `%.3f`.